lib/Backends/BackendUtils.cpp

```diff
@@ -55,3 +55,97 @@ size_t glow::getValueOffset(
+
+  for (auto &v : F.getGraph()->getParent()->getConstants()) {
+    assert(isa<WeightVar>(F.getWeightForNode(v)) && "Expected WeightVar");
+    auto *w = cast<WeightVar>(F.getWeightForNode(v));
+    auto numBytes = w->getSizeInBytes();
+    size_t addr = constantAllocator.allocate(numBytes, v);
+    runtime::RuntimeSymbolInfo symbol;
+    symbol.size = numBytes;
+    symbol.offset = addr;
+    symbol.type = *w->getType();
+    symbolTable.emplace(std::string(v->getName()), symbol);
+  }
+  auto constantMaxSize = constantAllocator.getMaxMemoryUsage();
+
+  // Compute the offsets for Placeholders.
+
+  for (auto &v : F.getGraph()->getParent()->getPlaceholders()) {
+    // Get the WeightVar for each Placeholder to calculate offsets.
+    assert(isa<WeightVar>(F.getWeightForNode(v)) && "Expected WeightVar");
+    auto *w = cast<WeightVar>(F.getWeightForNode(v));
+    auto numBytes = w->getSizeInBytes();
+    size_t addr = placeholderAllocator.allocate(numBytes, w);
+    runtime::RuntimeSymbolInfo symbol;
+    symbol.offset = addr;
+    symbol.size = numBytes;
+    symbol.type = *w->getType();
+    symbolTable.emplace(std::string(v->getName()), symbol);
+  }
+  auto placeholderMaxSize = placeholderAllocator.getMaxMemoryUsage();
+
+  // Compute the offsets for Activations.
+
+  for (const auto &I : F.getInstrs()) {
+    if (auto *A = llvm::dyn_cast<AllocActivationInst>(&I)) {
+      auto numBytes = I.getSizeInBytes();
+      size_t addr = activationsAllocator.allocate(numBytes, A);
+      assert(!symbolTable.count(std::string(A->getName())) &&
+             "Allocation already made!");
+      runtime::RuntimeSymbolInfo symbol;
+      symbol.offset = addr;
+      symbol.size = numBytes;
+      symbol.type = *A->getType();
+      symbolTable.emplace(std::string(A->getName()), symbol);
+      continue;
+    }
+
+    if (auto *TV = llvm::dyn_cast<TensorViewInst>(&I)) {
+      // Calculate and store the length of the offset into the base, using the
+      // source of the tensorview.
+      assert(!symbolTable.count(std::string(TV->getName())) &&
+             "Allocation already made!");
+      size_t offsetLength = TV->getOffsets().empty() ? 0 : TV->getOffsets()[0];
+      auto *tvSource = TV->getSrc();
+      if (tvSource->dims().size() > 1) {
+        for (size_t i = 1; i < tvSource->dims().size(); ++i) {
+          offsetLength *= tvSource->dims()[i];
+        }
+      }
+      assert(symbolTable.count(std::string(tvSource->getName())) &&
+             "Source allocation not found!");
+      runtime::RuntimeSymbolInfo symbol;
+      symbol.offset = getValueOffset(tvSource, symbolTable) +
+                      (offsetLength * TV->getType()->getElementSize());
+      symbol.size = TV->getSizeInBytes();
+      symbol.type = *TV->getType();
+      symbolTable.emplace(std::string(TV->getName()), symbol);
+      continue;
+    }
+    // Remove buffer from allocator if not needed at runtime.
+    if (auto *D = llvm::dyn_cast<DeallocActivationInst>(&I)) {
```

Comment:
The comment makes the understanding even more confusing :P.
If we don't need a buffer at runtime, why does it show up at all?

I believe that's not what you meant.